lumicks/pylake/scalebar.py

```diff
@@ -0,0 +1,153 @@
+from dataclasses import dataclass, field
+from matplotlib.offsetbox import AnchoredOffsetbox, AuxTransformBox, VPacker, HPacker, TextArea
+
+
+def _create_scale_legend(
+    transform,
+    size_x,
+    size_y,
+    label_x,
+    label_y,
+    loc,
+    color,
+    separation,
+    barwidth,
+    fontsize,
+    **kwargs,
+):
+    """Draws a scale legend for the current axis.
+
+    Parameters
+    ----------
+    transform : :class:`matplotlib.transforms.CompositeGenericTransform`
+        The coordinate frame (typically axes.transData).
+    size_x, size_y : float
+        Length of the scale bars in data units. `0` or `None` omits the scale.
+    label_x, label_y : str or None
+        Labels for the scale bars. None results in no label on the scale.
+    loc : position in the containing axis
+        Valid locations are 'upper left', 'upper center', 'upper right', 'center left',
+        'center', 'center right', 'lower left', 'lower center', 'lower right'.
+    color : color
```

Comment:
there should be the type here

Reply:
This is actually an interesting one. I wasn't sure what to put here. Basically, the type should be any color `matplotlib` can handle. When I look at [their](https://github.com/matplotlib/matplotlib/blob/c23ccdde6f0f8c071b09a88770e24452f2859e99/lib/matplotlib/colors.py#L271) [code](https://github.com/matplotlib/matplotlib/blob/c23ccdde6f0f8c071b09a88770e24452f2859e99/lib/matplotlib/lines.py#L1060), I see them just referring to these as `color` never specifying concrete types for them.

Currently, I think they are a union of `arraylike` and `str`, but to me, that seems less instructive to the user than `color`. It would also mean that if `mpl` supported additional types for their colors in the future, we would have to update this docstring.

What do you think?

Reply:
yeah that's an interesting one :D
looks like it can be a string or a tuple: https://matplotlib.org/stable/tutorials/colors/colors.html

 at this point I would put `Matplotlib color` as they say [here](https://github.com/matplotlib/matplotlib/blob/c23ccdde6f0f8c071b09a88770e24452f2859e99/lib/matplotlib/colors.py#L271)

Reply:
Yeah, I agree it's probably good to be explicit, since it's not our own definition.